Python/omevv/v1/scripts/LicenseImport.py

```diff
@@ -0,0 +1,199 @@
+        payload["ShareType"] = args["sharetype"]
+    if args["sharename"]:
+        payload["ShareName"] = args["sharename"]
+    if args["username"]:
+        payload["UserName"] = args["username"]
+    if args["password"]:
+        payload["Password"] = args["password"]
+    if args["workgroup"]:
+        payload["Workgroup"] = args["workgroup"]
+    if args["ignorecertwarning"]:
+        payload["IgnoreCertificateWarning"] = args["ignorecertwarning"]
+    if args["retry"]:
+        retry = args["retry"]
+
+    print("\n- WARNING, arguments and values for License Import method\n")
+    for i in payload.items():
+        if i[0] == "ShareParameters":
+            for ii in i[1].items():
+                if ii[0] == "Password":
+                    print("Password: **********")
+                else:
+                    print("%s: %s" % (ii[0], ii[1]))
+        else:
+            print("%s: %s" % (i[0], i[1]))
+
+
+def import_idrac_license_network_share():
+    global retry, job_id
+    try:
+        headers = {'content-type': 'application/json'}
+        method = "ImportLicenseFromNetworkShare"
+        url = 'https://%s/redfish/v1/Dell/Managers/iDRAC.Embedded.1/DellLicenseManagementService/Actions/DellLicenseManagementService.ImportLicenseFromNetworkShare' % (
+            idrac_ip)
+
+        response = requests.post(url, data=json.dumps(payload), headers=headers, verify=False,
+                                 auth=(idrac_username, idrac_password))
+        data = response.json()
+        status_code = response.status_code
+        if response.status_code == 202:
+            print("\n- PASS: POST command passed for %s method, status code %s returned\n" % (
+            method, response.status_code))
+        # 500 series status codes
+        elif response.status_code >= 500:
+            print("\n- FAIL:POST command failed for %s method, status code %s.Retry after sometime\n" % (
+            method, status_code))
+            print("\n- POST command failure results:\n %s" % data)
+            sys.exit()
+        # 400 series status codes
+        else:
+            print("\n- FAIL: POST command failed for %s method, status code %s\n" % (method, status_code))
+            print("\n- POST command failure results:\n %s" % data)
+            sys.exit()
+
+        job_id = response.headers.get('Location')
+        if not (job_id):
+            print("- FAIL, unable to find job ID in headers POST response, headers output is:\n%s" % response.headers)
+            sys.exit()
+        print("- PASS, job ID %s successfuly created for %s method\n" % (job_id, method))
+
+    except Exception as e:
+        print("\n- FAIL: Exception %s occured" % e)
+        if retry > 0:
+            retry = retry - 1
+            time.sleep(5)
+            import_idrac_license_network_share()
+        else:
+            print("- FAIL: Exception %s occured even after retrying %s times" % (e, retry))
+            sys.exit()
+
+
+def loop_job_status():
```

Comment:
If this is a seperate method, then the the previous method must return the job Id that can be polled. This method can then take the job id as a parameter and return the status that can be routed to the console in the main method and handled appropriately by any other script using this as a module.

Reply:
added job_id as input to the method